delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/model/StorageValue.scala

```diff
@@ -0,0 +1,87 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.model
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+
+import java.nio.file.Path
+
+sealed trait StorageValue extends Product with Serializable {
+
+  /**
+    * @return the storage type
+    */
+  def tpe: StorageType
+
+}
+
+object StorageValue {
+
+  /**
+    * Necessary values to create/update a disk storage
+    *
+    * @param default         ''true'' if this store is the project's default backend, ''false'' otherwise
+    * @param algorithm       the digest algorithm
+    * @param volume          the volume this storage is going to use to save files
+    * @param readPermission  the permission required in order to download a file from this storage
+    * @param writePermission the permission required in order to upload a file to this storage
+    * @param maxFileSize     the maximum allowed file size (in bytes) for uploaded files
+    */
+  final case class DiskStorageValue(
+      default: Boolean,
+      algorithm: Algorithm,
+      volume: Path,
+      readPermission: Permission,
+      writePermission: Permission,
+      maxFileSize: Long
```

Comment:
We could have a refined type to make sure it is positive and also to put an upper-value that is reasonable

Reply:
The upper value is driven by configuration. So since we have to check dynamically for the upper value, I would rather check for both than having a custom case class for this.

Reply:
So this check could be part of the STM logic no ?